lavis/datasets/datasets/caption_wpi_datasets.py

```diff
@@ -0,0 +1,68 @@
+import os
+import pandas as pd
+from torchvision.io import read_image
+from torch.utils.data import Dataset
+import tqdm
+from PIL import Image
+
+
+class WpiDataset(Dataset):
+    def __init__(
+            self,
+            vis_processor=None,
+            text_processor=None,
+            vis_root=None,
+            ann_paths=[],
+            label_name='Topics'):
+        """
+        Load annotations - Each annotation consists of:
+        - otional: genres (e.g. ["notes (documents)"]), topics, places
+        - required: title, image path
+        - Label group for dataset reference is set with param label_name
+        """
+        annotations_file = ann_paths[0]
+        self.anno_df = pd.read_json(annotations_file)
+        self.img_dir = vis_root
+        self.label_name = label_name
+        self.labels_array = self.labels()
+        self.vis_processor = vis_processor
+        self.text_processor = text_processor
+        if label_name == 'Genres':
+            self.label_idx = 0
+        elif label_name == 'Topics':
+            self.label_idx = 5
+        elif label_name == 'Places':
+            self.label_idx = 6
+        else:
+            raise ValueError(
+                'Must supply a valid label_name. Options are: Genres, Topics or Places')
+
+    def labels(self):
+        occurrences = dict()
+        for _, labels in self.anno_df[self.label_name].dropna().items():
```

Comment:
use `.values()` if you don't need the keys

Reply:
done

Reply:
done